src/mpi/comm/comm_split_type.c

```diff
@@ -10,6 +10,15 @@
 
 #ifdef HAVE_HWLOC
 #include "hwloc.h"
+#ifdef HWLOC_HAVE_CUDA
+#include "hwloc/cuda.h"
+#endif
+#ifdef HWLOC_HAVE_OPENCL
+#include "hwloc/opencl.h"
+#endif
+#ifdef HWLOC_HAVE_LIBIBVERBS_TRUE
+#include "hwloc/openfabrics-verbs.h"
+#endif
```

Comment:
Do we really need to explicitly add these?  Aren't these available in the public header?

Reply:
The hwloc headers include cuda header which may or may not be available. I thought that conditionally including the headers would build on all platforms.

Reply:
I wasn't suggesting adding them unconditionally.  Why isn't the hwloc public header conditionally including them?

Reply:
I will check this with hwloc dev guys and get back to you on this.

Reply:
Why?  You have the source code.

Reply:
I know they add cl and cuda headers unconditionally, I don't know why that is. I'll check this and get back to you.